react-native/react/login2/register/set-public-name/index.js

```diff
@@ -0,0 +1,50 @@
+'use strict'
+/* @flow */
+
+import React, {Component} from '../../../base-react'
+import {setDeviceName} from '../../../actions/login2'
+import Render from './index.render'
+
+export default class SetPublicName extends Component {
+  nameTaken (deviceName) {
+    return this.props.existingDevices && this.props.existingDevices.indexOf(deviceName) !== -1
+  }
+
+  submitEnabled (deviceName) {
+    return deviceName && deviceName.length && !this.nameTaken(deviceName)
+  }
+
+  render () {
+    return (
+      <Render
+        onSubmit={ deviceName => this.props.dispatch(setDeviceName(deviceName)) }
+        nameTaken={ deviceName => this.nameTaken(deviceName) }
+        submitEnabled={ deviceName => this.submitEnabled(deviceName) }
+        {...this.props}
+      />
+    )
+  }
+
+  static parseRoute (store, currentPath, nextPath) {
+    return {
+      componentAtTop: {
+        title: '',
+        component: SetPublicName,
+        leftButtonTitle: '',
+        mapStateToProps: state => {
```

Comment:
Totally unrelated to this PR, just out of curiosity: why do we use this instead of exporting a `connect`ed version of the component like all of the Redux sample code?